### ISSUE TITLE
Limitation in mapping.json: case insensitivity

### ISSUE THREAD

**Issue by homersimpsons** [exercism/php-representer#27]
In PHP some tokens are case-insensitive, namely:
- Functions names: https://www.php.net/manual/en/functions.user-defined.php
  > Note: Function names are case-insensitive for the ASCII characters A to Z, though it is usually good form to call functions as they appear in their declaration. 
- class names
- methods (functions in classes)

The current `mapping.json` design allows to map a string from the "canonical" `representation.txt` to a single value from the original source.

With PHP's case insensitivity this is not enough, in fact given the following code:

```php
<?php
function hElLoWoRlD() {}
HeLlOwOrLd();
```

our `reprensentation.txt` file will look like:
```
function f0() {}
f0();
```

and our `mapping.json` file will look like:
```
{"fn0":"helloworld"}
```

What can we do about those cases @ErikSchierboom ?

We could upgrade the `mapping.json` structure to allow something like:
```
{"fn0":["hElLoWoRlD","HeLlOwOrLd"]}
```

**Comment by ErikSchierboom** [comment]
```
{"fn0":"helloworld"}
```

Why does it look like this and not like

```
{"fn0":"hElLoWoRlD"}
```

**Comment by homersimpsons** [comment]
Because I normalize the name, but even if I represent it like:

```
{"fn0":"hElLoWoRlD"}
```

There still is the issue that `"HeLlOwOrLd"` is not part of the `mapping.json` while it still is mapped.

**Comment by ErikSchierboom** [comment]
Is that because when you analyze the code you no longer have access to the original text?

**Comment by homersimpsons** [comment]
> Is that because when you analyze the code you no longer have access to the original text?

No, I still have the original text, in fact I do those transformations here:

https://github.com/exercism/php-representer/blob/9b99149f5d58ba8eaabc70572b62150d1c057594/src/Mapping.php#L89

https://github.com/exercism/php-representer/blob/9b99149f5d58ba8eaabc70572b62150d1c057594/src/Mapping.php#L121

https://github.com/exercism/php-representer/blob/9b99149f5d58ba8eaabc70572b62150d1c057594/src/Mapping.php#L133

But as I said in the original issue, PHP is case insensitive on those tokens, so even if I keep `HeLlOwOrLd` there will still be an issue with `hElLoWoRlD` which on PHP side are considered the same.

May be this is acceptable, I do not know what is exactly the purpose of the `mapping.json` file. If it is to generate feedback and it just expands the mapping into the text this might be sufficient.

**Comment by ErikSchierboom** [comment]
>  If it is to generate feedback and it just expands the mapping into the text this might be sufficient.

It is exactly this.

**Comment by homersimpsons** [comment]
> > If it is to generate feedback and it just expands the mapping into the text this might be sufficient.
> 
> It is exactly this.

Okay perfect, I will then just track the "most common" mapping value so it should feel consistent enough for the user.

Alternatively I could also write the mapping as `{"fn0":"hElLoWoRlD (HeLlOwOrLd)"}` do you think that would make sense ?
